### ISSUE TITLE
IQKeyboardManager - quicktype + iPhoneSE

### ISSUE THREAD

**Issue by pgawlowski** [hackiftekhar/IQKeyboardManager#1027]
Two problems probably connected:
- first quicktype toolbar 
![img_0529](https://user-images.githubusercontent.com/9845024/30967579-0b3549e0-a45d-11e7-866b-0bb618d5bd11.jpg)

It's covering e-mail textField in my case. It's probably caused by the way keyboard is generating quicktype toolbar - first it's showing keyboard then it's showing quicktype and it's causing overlaying textField.

Second problem I noticed on iPhoneSE (both simulator and device)
Keyboard avoiding is not working in a correct like - totally, even applying keyboardDistanceFromTextField is not giving any effect.

It's moving just a little bit but again toolbar is overlaying textfield (this time without quicktype).

<img width="302" alt="screen shot 2017-09-28 at 14 59 52" src="https://user-images.githubusercontent.com/9845024/30967871-c05e9498-a45d-11e7-8ead-cdb5fab372b6.png">
<img width="328" alt="screen shot 2017-09-28 at 15 00 03" src="https://user-images.githubusercontent.com/9845024/30967873-c20dac02-a45d-11e7-8e11-0338c1e6b2fc.png">

**Comment by hackiftekhar** [comment]
@pgawlowski Can you please check if you pinnged your textField with layout guide or safe area?

**Comment by pgawlowski** [comment]
I'm not sure If I understand. UITextField is a subview of another view. Here is a screen of hierarchy (higlighted one is my textField) + constraints.
<img width="629" alt="screen shot 2017-09-28 at 15 12 16" src="https://user-images.githubusercontent.com/9845024/30968418-8e14a6b0-a45f-11e7-9326-3dabfab3aae1.png">

**Comment by hackiftekhar** [comment]
Now I can see more clearly. @pgawlowski Can you please check your UITextField container UIView(just above your selection) pinned with it's superview or with Top Layout guide?

**Comment by pgawlowski** [comment]
@hackiftekhar It's pinned with another object (logo).

What about first part of my question? I mean with e-mail field and quicktype?

**Comment by hackiftekhar** [comment]
The quicktype toolbar depends on user settings, this isn't added by library, it does affect with a Device Settings and the autocorrect property of UITextField. You should change it to hide it.

**Comment by hackiftekhar** [comment]
Regarding main issue, have your logo pinned to top layout guide constraint. Technically I mean please check if your top layout guide or bottom layout guide affect your textfield position directly or indirectly.

**Comment by pgawlowski** [comment]
It is. From top:
- logo - pinned to topLayoutGuide
- label
- another label
- view with: - label, problematic textField

**Comment by pgawlowski** [comment]
Ok, both problems seems to be strictly connected with top constraint. 
By default I had top element constraint set to TopLayoutGuide it was causing the wrong paddings whole time.

I forced to connect top constraint to superview and it started to work again. However - right now interface builder by default is being added to LayoutGuides not to superview. In demo you are probably using a little bit older storyboards. Try to set constraints in demo once again - by default it will be added to TopLayoutGuides and it will probably cause problems I described.
